source/material_model/grain_size.cc

```diff
@@ -979,10 +978,11 @@ namespace aspect
           phase_function->initialize_simulator (this->get_simulator());
           phase_function->parse_parameters (prm);
 
-          std::vector<unsigned int> n_phases_for_each_composition = phase_function->n_phases_for_each_composition();
-          n_phase_transitions = n_phases_for_each_composition[0] - 1;
+          // Phase transitions are only supported for the background composition.
```

Comment:
Not sure that's quite the right way to say it. There is not really a background composition. The phase function is only used for the viscosity, and you can't specify different viscosities for the different compositions. So technically, the phase transitions are _the same_ for all compositions.